src/orders/views/OrderDetails/OrderNormalDetails/index.tsx

```diff
@@ -115,29 +123,30 @@ export const OrderNormalDetails: React.FC<OrderNormalDetailsProps> = ({
   >(null);
 
   React.useEffect(() => {
-    const warehousesAvailability = warehouses?.map(warehouse => {
-      if (!order?.lines) {
-        return undefined;
-      }
+    const warehousesAvailability: WarehousesAvailibility[] = warehouses?.map(
+      warehouse => {
+        if (!order?.lines) {
+          return undefined;
+        }
+        const linesToFulfill = getToFulfillOrderLines(order.lines);
 
-      const linesAvailable = order.lines.filter(line =>
-        line?.variant?.stocks?.find(
-          stock => stock.warehouse.id === warehouse.id
-        )
-      ).length;
+        const linesAvailable = linesToFulfill.filter(line =>
+          isLineAvailableInWarehouse(line, warehouse)
+        ).length;
 
-      return {
-        warehouse,
-        linesAvailable
-      };
-    });
+        return {
+          warehouse,
+          linesAvailable
+        };
+      }
+    );
```

Comment:
This file is already cluttered. Can we make is a separate hook, or maybe extract map function to util file?